scripts/test_smoke.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+# Run smoke tests to test that updating between versions work.
+#
+# This is based on our update tests but doing some tweaks to ensure we
+# can run it on Forge (or any other PostgreSQL server that only permit
+# a single user and single database).
+#
+# In particular, we cannot create new roles and we cannot create new
+# databases.
+#
+# For this version, we focus on single-node update and smoke testing
+# cannot (necessarily) be done with multinode setups. Feel free to try
+# though.
+#
+# The following environment variables can be set: 
+# - UPDATE_FROM_TAG is the version to update from (optional).
+#
+# - UPDATE_TO_TAG is the version to update to (optional).
+#
+# - PGHOST is host to use for the connection (required).
+#
+# - PGPORT is the port to use for the connection (required).
+#
+# - PGDATABASE is the database to use for the connection (required).
+#
+# - PGUSER is the username to use for the connection (required).
+#
+# - PGPASSWORD is the password to use for the connection
+#   (optional). If not set, password from .pgpass will be used (if
+#   available).
+
+: ${PGUSER:?Please provide the user in PGUSER}
+: ${PGHOST:?Please provide the host in PGHOST}
+: ${PGPORT:?Please provide the port in PGPORT}
+: ${PGDATABASE:?Please provide the database in PGDATABASE}
+
+cleanup() {
+    rm -rf $SCRATCHDIR
+}
+
+while getopts "d" opt;
+do
+    case $opt in
+        d)
+            cleanup=1
+            ;;
+	r)
+	    TEST_REPAIR=true
+	    ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+SCRIPT_DIR=$(dirname $0)
+BASE_DIR=${PWD}/${SCRIPT_DIR}/..
+SCRATCHDIR=`mktemp -d -t 'smoketest-XXXX'`
+DUMPFILE="$SCRATCHDIR/smoke.dump"
+UPGRADE_OUT="$SCRATCHDIR/upgrade.out"
+CLEAN_OUT="$SCRATCHDIR/clean.out"
+RESTORE_OUT="$SCRATCHDIR/restore.out"
+UPDATE_FROM_TAG=${1:-${UPDATE_FROM_TAG:-1.7.4}}
+UPDATE_TO_TAG=${2:-${UPDATE_TO_TAG:-2.0.1}}
+TEST_REPAIR=false
+TEST_VERSION=${TEST_VERSION:-v6}
+
+# Extra options to pass to psql
+PGOPTS="-v TEST_VERSION=${TEST_VERSION} -v TEST_REPAIR=${TEST_REPAIR}"
+PSQL="psql -X $PGOPTS"
+
+if [[ "x$cleanup" != "x" ]]; then
+    echo "**** Debug mode: $SCRATCHDIR will not be removed"
+    trap cleanup EXIT
+fi
+
+set -e
+
+echo "**** Scratch directory: ${SCRATCHDIR}"
+echo "**** Update files in directory ${BASE_DIR}/test/sql/updates"
+cd ${BASE_DIR}/test/sql/updates
+
+echo "**** Connecting as user '$PGUSER' to database $PGDATABASE on $PGHOST:$PGPORT"
+
+$PSQL -qc "SELECT name, version FROM pg_available_extension_versions WHERE name = 'timescaledb' AND version IN ('$UPDATE_FROM_TAG', '$UPDATE_TO_TAG')"
+
+AVAIL=$(
+    $PSQL --csv -tqc "SELECT COUNT(*) FROM pg_available_extension_versions WHERE name = 'timescaledb' AND version IN ('$UPDATE_FROM_TAG', '$UPDATE_TO_TAG')"
+	)
+
+if [[ $AVAIL != 2 ]]; then
+    echo "Missing a version"
```

Comment:
would be useful to add both version names here. Along the lines of: "Missing a version. Either version x or version y is not installed"/

Reply:
Added some code that printed the missing versions, if there are any.